src/support_enumeration.jl

```diff
@@ -0,0 +1,293 @@
+                                         supps[1], supps[2])
+                    if _indiff_mixed_action!(A, b, actions[1],
+                                             payoff_matrices[2],
+                                             supps[2], supps[1])
+                        out = (zeros(S, nums_actions[1]),
+                               zeros(S, nums_actions[2]))
+                        for (p, (supp, action)) in enumerate(zip(supps,
+                                                                 actions))
+                            out[p][supp] = action
+                        end
+                        produce(out)
+                    end
+                end
+                _next_k_array!(supps[2])
+            end
+            _next_k_array!(supps[1])
+        end
+    end
+
+end
+
+"""
+    _indiff_mixed_action!{T<:Real}(A::Matrix{T}, b::Vector{T},
+                                   out::Vector{T},
+                                   payoff_matrix::Matrix{T},
+                                   own_supp::Vector{Int},
+                                   opp_supp::Vector{Int})
+
+Given a player's payoff matrix `payoff_matrix`, an array `own_supp`
+of this player's actions, and an array `opp_supp` of the opponent's
+actions, each of length k, compute the opponent's mixed action whose
+support equals `opp_supp` and for which the player is indifferent
+among the actions in `own_supp`, if any such exists. Return `true`
+if such a mixed action exists and actions in `own_supp` are indeed
+best responses to it, in which case the outcome is stored in `out`;
+`false` otherwise. Arrays `A` and `b` are used in intermediate
+steps.
+
+# Arguments
+
+* `A::Matrix{T}`: Matrix used in intermediate steps.
+* `b::Vector{T}`: Vector used in intermediate steps.
+* `out::Vector{T}`: Vector to store the nonzero values of the
+  desired mixed action.
+* `payoff_matrix::Matrix{T}`: The player's payoff matrix.
+* `own_supp::Vector{Int}`: Vector containing the player's action indices.
+* `opp_supp::Vector{Int}`: Vector containing the opponent's action indices.
+
+# Returns
+
+* `::Bool`: `true` if a desired mixed action exists and `false` otherwise.
+"""
+function _indiff_mixed_action!{T<:Real}(A::Matrix{T}, b::Vector{T},
+                                        out::Vector{T},
+                                        payoff_matrix::Matrix{T},
+                                        own_supp::Vector{Int},
+                                        opp_supp::Vector{Int})
+
+    m = size(payoff_matrix, 1)
+    k = length(own_supp)
+
+    A[1:end-1, 1:end-1] = payoff_matrix[own_supp, :][:, opp_supp]
+    A[1:end-1, end] = -one(T)
+    A[end, 1:end-1] = one(T)
+    A[end, end] = zero(T)
+    b[1:end-1] = zeros(T, k)
+    b[end] = one(T)
+    try
+        b = A_ldiv_B!(lufact!(A), b)
+    catch LinAlg.SingularException
+        return false
+    end
+
+    if any(b[1:end-1] .<= 0)
+        return false
+    end
```

Comment:
What about

```julia
if findfirst(x -> x<=zero(T), b) != 0
    return false
end
```

According to the [source code](https://github.com/JuliaLang/julia/blob/f45988f76a45549e7ef33a8377a4ad8f96c0d9ef/base/array.jl#L1240-L1258) of `findfirst`, it is doing exactly what your code does, and I am wondering if it would be better to use the function that is available in Base.